tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
Probably a bug in watcher. can you try to debug it a little bit?

/cc @heyitsanthony